pymeasure/experiment/results.py

```diff
@@ -269,7 +269,7 @@ def units_found(parameter, units):
             if parameter.name in parameters:
                 value, units = parameters[parameter.name]
 
-                if units_found(parameter, units):
+                if units_found(parameter, units) or (type(parameter) == Parameter and units is not None):
```

Comment:
In my fix for the string with space bug I simply extended the if-statement that was already there. I wonder though what the purpose was for the original if-statement, since I couldn't find a case where this was True, nor could think of a case where this should be True. Do you have an idea why it was there @cjermain (a variant of this seems to have been there since your first commit)?